## 내 구글 드라이브 마운팅
본인 구글 내 데이터 사용

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


## 선수정보, 연봉, 타자, 투수, 팀 데이터 읽어오기

In [0]:
import pandas as pd
player=pd.read_csv('/content/drive/My Drive/통계적방법19_박초연/People.csv')
salary=pd.read_csv('/content/drive/My Drive/통계적방법19_박초연/Salaries.csv')
batter=pd.read_csv('/content/drive/My Drive/통계적방법19_박초연/Batting.csv')
pitcher=pd.read_csv('/content/drive/My Drive/통계적방법19_박초연/Pitching.csv')
team=pd.read_csv('/content/drive/My Drive/통계적방법19_박초연/Teams.csv')

## 만든 데이터프레임 변수명 출력

In [3]:
print('선수정보',player.columns,'\n연봉',salary.columns,
'\n타자',batter.columns,'\n투수',pitcher.columns,'\n팀',team.columns)

선수정보 Index(['playerID', 'birthYear', 'birthMonth', 'birthDay', 'birthCountry',
       'birthState', 'birthCity', 'deathYear', 'deathMonth', 'deathDay',
       'deathCountry', 'deathState', 'deathCity', 'nameFirst', 'nameLast',
       'nameGiven', 'weight', 'height', 'bats', 'throws', 'debut', 'finalGame',
       'retroID', 'bbrefID'],
      dtype='object') 
연봉 Index(['yearID', 'teamID', 'lgID', 'playerID', 'salary'], dtype='object') 
타자 Index(['playerID', 'yearID', 'stint', 'teamID', 'lgID', 'G', 'AB', 'R', 'H',
       '2B', '3B', 'HR', 'RBI', 'SB', 'CS', 'BB', 'SO', 'IBB', 'HBP', 'SH',
       'SF', 'GIDP'],
      dtype='object') 
투수 Index(['playerID', 'yearID', 'stint', 'teamID', 'lgID', 'W', 'L', 'G', 'GS',
       'CG', 'SHO', 'SV', 'IPouts', 'H', 'ER', 'HR', 'BB', 'SO', 'BAOpp',
       'ERA', 'IBB', 'WP', 'HBP', 'BK', 'BFP', 'GF', 'R', 'SH', 'SF', 'GIDP'],
      dtype='object') 
팀 Index(['yearID', 'lgID', 'teamID', 'franchID', 'divID', 'Rank', 'G', 'Ghome',
       'W', 'L', 'DivWin'

## 만든 데이터프레임 크기 보기

In [4]:
print('선수정보',player.shape,'\n연봉',salary.shape,'\n타자',batter.shape,
'\n투수',pitcher.shape,'\n팀',team.shape)

선수정보 (19617, 24) 
연봉 (26428, 5) 
타자 (105861, 22) 
투수 (46699, 30) 
팀 (2895, 48)


## 선수정보 + 연봉정보 합치기 (선수_연봉)
합친 데이터프레임 이름 : (선수_연봉) ps

In [5]:
ps=pd.merge(salary,player,on='playerID',how='inner')
ps.sort_values(['yearID','playerID'],inplace=True)
print(ps.shape,'\n',ps.columns)

(26428, 28) 
 Index(['yearID', 'teamID', 'lgID', 'playerID', 'salary', 'birthYear',
       'birthMonth', 'birthDay', 'birthCountry', 'birthState', 'birthCity',
       'deathYear', 'deathMonth', 'deathDay', 'deathCountry', 'deathState',
       'deathCity', 'nameFirst', 'nameLast', 'nameGiven', 'weight', 'height',
       'bats', 'throws', 'debut', 'finalGame', 'retroID', 'bbrefID'],
      dtype='object')


## 타자 정보 + 선수_연봉정보 합치기

In [6]:
bat=pd.merge(ps,batter,on=['yearID','playerID'],how='inner')
print(bat.shape,'\n',bat.columns)

(28294, 48) 
 Index(['yearID', 'teamID_x', 'lgID_x', 'playerID', 'salary', 'birthYear',
       'birthMonth', 'birthDay', 'birthCountry', 'birthState', 'birthCity',
       'deathYear', 'deathMonth', 'deathDay', 'deathCountry', 'deathState',
       'deathCity', 'nameFirst', 'nameLast', 'nameGiven', 'weight', 'height',
       'bats', 'throws', 'debut', 'finalGame', 'retroID', 'bbrefID', 'stint',
       'teamID_y', 'lgID_y', 'G', 'AB', 'R', 'H', '2B', '3B', 'HR', 'RBI',
       'SB', 'CS', 'BB', 'SO', 'IBB', 'HBP', 'SH', 'SF', 'GIDP'],
      dtype='object')


## 투수 정보 + 선수_연봉정보 합치기

In [7]:
pitch=pd.merge(ps,pitcher,on=['yearID','playerID'],how='inner')
print(pitch.shape,'\n',pitch.columns)

(13365, 56) 
 Index(['yearID', 'teamID_x', 'lgID_x', 'playerID', 'salary', 'birthYear',
       'birthMonth', 'birthDay', 'birthCountry', 'birthState', 'birthCity',
       'deathYear', 'deathMonth', 'deathDay', 'deathCountry', 'deathState',
       'deathCity', 'nameFirst', 'nameLast', 'nameGiven', 'weight', 'height',
       'bats', 'throws', 'debut', 'finalGame', 'retroID', 'bbrefID', 'stint',
       'teamID_y', 'lgID_y', 'W', 'L', 'G', 'GS', 'CG', 'SHO', 'SV', 'IPouts',
       'H', 'ER', 'HR', 'BB', 'SO', 'BAOpp', 'ERA', 'IBB', 'WP', 'HBP', 'BK',
       'BFP', 'GF', 'R', 'SH', 'SF', 'GIDP'],
      dtype='object')


In [8]:
pitch.head(10)

yearID teamID_x lgID_x   playerID   salary  ...  GF    R    SH   SF  GIDP
0    1985      TOR     AL  ackerji01   170000  ...  26   35   1.0  2.0  16.0
1    1985      CHA     AL  agostju01   147500  ...  21   27   3.0  3.0   4.0
2    1985      TOR     AL  alexado01   875000  ...   0  105   6.0  3.0  15.0
3    1985      SLN     NL  allenne01   750000  ...  13   22   1.0  3.0   3.0
4    1985      SLN     NL  allenne01   750000  ...  10    9   0.0  0.0   3.0
5    1985      PHI     NL  anderla02   250500  ...  19   41   3.0  1.0   4.0
6    1985      SLN     NL  andujjo01  1030000  ...   0  113  11.0  4.0  32.0
7    1985      OAK     AL  atherke01   107333  ...  21   51   3.0  4.0   4.0
8    1985      CHA     AL  bannifl01   811250  ...   0  121   9.0  8.0  20.0
9    1985      ATL     NL  barkele01   870000  ...   1   55   4.0  1.0   4.0

[10 rows x 56 columns]

## [HW#1 due 내일 수업전까지]
bat, pitch 데이터프레임에서 stint=2인 선수(시즌 중 이적한 선수임)를 제외하는 코드를 작성하시오.

In [9]:
bat.shape

(28294, 48)

[참고] 타자_선수_연봉 데이터 프레임 - stint 카운트 하면
2= 시즌 중 2번 이적, 3= 시즌 중 3번 이적, 4= 시준 중 4번 이적 그러므로 2,3,4인 선수는 1도 제외하여함

In [10]:
bat['stint'].value_counts()

1    25603
2     2567
3      117
4        7
Name: stint, dtype: int64

예상되는 타자 데이터프레임 행 크기 => 28,294에서 22912

In [11]:
25603-2567-117-7

22912

In [12]:
bat[['playerID','stint']][27415:27425]

playerID  stint
27415  archech01      1
27416  arciaos01      1
27417  arciaos01      2
27418  arciaos01      3
27419  arciaos01      4
27420  arenano01      1
27421  arrieja01      1
27422  ascheco01      1
27423  avilaal01      1
27424  avilemi01      1

In [13]:
bat2 = bat.drop_duplicates(['playerID','yearID'], keep=False)
bat2.shape

(22964, 48)

In [14]:
bat2[['playerID','stint']][22290:22300]

playerID  stint
27462  blackch02      1
27463  blancan01      1
27464  blancgr01      1
27465  blantjo01      1
27466  blashja01      1
27467  blazemi01      1
27468  blevije01      1
27469  bogaexa01      1
27470  bolsimi01      1
27471  bourjpe01      1

stint의 변수 중 시즌 중 이동이 있음에도 1로 기입된 선수 36명에 대해 처리 불가능!

In [15]:
pitch[['playerID','stint']][7415:7425]

playerID  stint
7415  hudsoti01      1
7416  ishiika01      1
7417  isrinja01      1
7418  jarvike01      1
7419  jennija01      1
7420  jensery01      1
7421  jimenjo01      1
7422  johnsja02      1
7423  johnsra05      1
7424  jonesto02      1

In [16]:
pitch2 = pitch.drop_duplicates(['playerID','yearID'], keep=False)
pitch2 = pitch2[(pitch2.stint==1)]
pitch2.shape

(10730, 56)

##문제1. 2016년 연봉평균은 2015년 연봉평균보다 올랐는가?
(이상치 제거X, 동질성 검정O)

>귀무가설(H0) : 2016년 연봉 평균(Mu1)은 2015년 연봉 평균(Mu2)과 같다. Mu1=Mu2

>대립가설(H1) : 2016년 연봉 평균(Mu1)은 2015년 연봉 평균(Mu2)보다 높다. Mu1hat>Mu2hat

유의수준:0.05

In [0]:
salary15=salary[(salary.yearID==2015)]
salary16=salary[(salary.yearID==2016)]

In [18]:
ms15=salary15.salary.mean()
ms16=salary16.salary.mean()
print(ms15,'\n',ms16)

4301276.094247246 
 4396409.603751466


분산 동질성 검정

>귀무가설(H0) : 2016년 연봉분산(V(X1))은 2015년 연봉분산(V(X2))과 같다. Mu1=Mu2 V(X1)=V(X2)

>대립가설(H1):2016년 연봉 분산(V(X1))은 2015년 연봉 분산V(X2)과 다르다. V(X1)!=V(X2)


-분산이 같은경우(귀무가설채택=pvalue가 크다) : stats.ttest_ind(data1,data2,equal_var=True)

-분산이 다를경우(귀무가설기각=pvalue가 작다) : stats.ttest_ind(data1,data2,equal_var=False)

(pvalue가 크다 작다는 유의수준으로 판단, 언급없으면 0.05)

In [19]:
import scipy.stats
scipy.stats.levene(salary16.salary,salary15.salary)

LeveneResult(statistic=0.49067508497863344, pvalue=0.48372401252620556)

>결론 : pvalue가 0.48로 유의수준 0.05보다 큰값을 보여, 귀무가설 채택
즉, 두 집단의 모분산은 같다.

In [22]:
import scipy.stats
scipy.stats.ttest_ind(salary16.salary,salary15.salary,equal_var=True)

Ttest_indResult(statistic=0.34030436064248343, pvalue=0.7336701842906231)

결론 : 분산이 같을 때, 두 평균이 같을 확률은 0.73으로 매우 크며 유의수준 0.05보다 커, 0.75의 높은 확률로 관측되었기 때문에 H0 연봉 평균이 같다는 가설이 맞다고 판단하여 귀무가설을 채택하며, 연봉 평균은 0.34일 것이다.